nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/endpoints/ListFlowFilesEndpointMerger.java

```diff
@@ -0,0 +1,158 @@
+
+    @Override
+    public boolean canHandle(URI uri, String method) {
+        if (("GET".equalsIgnoreCase(method) || "DELETE".equalsIgnoreCase(method)) && LISTING_REQUEST_URI.matcher(uri.getPath()).matches()) {
+            return true;
+        } else if ("POST".equalsIgnoreCase(method) && LISTING_REQUESTS_URI.matcher(uri.getPath()).matches()) {
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    protected Class<ListingRequestEntity> getEntityClass() {
+        return ListingRequestEntity.class;
+    }
+
+    @Override
+    protected ListingRequestDTO getDto(ListingRequestEntity entity) {
+        return entity.getListingRequest();
+    }
+
+    @Override
+    protected void mergeResponses(ListingRequestDTO clientDto, Map<NodeIdentifier, ListingRequestDTO> dtoMap, Set<NodeResponse> successfulResponses, Set<NodeResponse> problematicResponses) {
+        final Comparator<FlowFileSummaryDTO> comparator = new Comparator<FlowFileSummaryDTO>() {
+            @Override
+            public int compare(final FlowFileSummaryDTO dto1, final FlowFileSummaryDTO dto2) {
+                int positionCompare = dto1.getPosition().compareTo(dto2.getPosition());
+                if (positionCompare != 0) {
+                    return positionCompare;
+                }
+
+                final String address1 = dto1.getClusterNodeAddress();
+                final String address2 = dto2.getClusterNodeAddress();
+                if (address1 == null && address2 == null) {
+                    return 0;
+                }
+                if (address1 == null) {
+                    return 1;
+                }
+                if (address2 == null) {
+                    return -1;
+                }
+                return address1.compareTo(address2);
+            }
+        };
+
+        final NavigableSet<FlowFileSummaryDTO> flowFileSummaries = new TreeSet<>(comparator);
+
+        ListFlowFileState state = null;
+        int numStepsCompleted = 0;
+        int numStepsTotal = 0;
+        int objectCount = 0;
+        long byteCount = 0;
+        boolean finished = true;
+        for (final Map.Entry<NodeIdentifier, ListingRequestDTO> entry : dtoMap.entrySet()) {
+            final NodeIdentifier nodeIdentifier = entry.getKey();
+            final String nodeAddress = nodeIdentifier.getApiAddress() + ":" + nodeIdentifier.getApiPort();
+
+            final ListingRequestDTO nodeRequest = entry.getValue();
+
+            numStepsTotal++;
+            if (Boolean.TRUE.equals(nodeRequest.getFinished())) {
+                numStepsCompleted++;
+            }
+
+            final QueueSizeDTO nodeQueueSize = nodeRequest.getQueueSize();
+            objectCount += nodeQueueSize.getObjectCount();
+            byteCount += nodeQueueSize.getByteCount();
+
+            if (!nodeRequest.getFinished()) {
+                finished = false;
+            }
```

Comment:
Couple of questions here
1. Should this just be ```finished = nodeRequest.getFinished()```
2. It seems like you are setting it while iterating, but then using its value outside the loop (line:152). So I am assuming that one node can be finished and another is not and the value of 'finished' really depends on the order of iteration. Is that intentional?